dashboard/app/models/script.rb

```diff
@@ -799,17 +809,33 @@ def copy_and_write_i18n(new_name)
     File.write(scripts_yml, "# Autogenerated scripts locale file.\n" + i18n.to_yaml(line_width: -1))
   end
 
-  # script is found/created by 'id' (if provided) otherwise by 'name'
+  # script is found/created by 'id' (if provided), or by 'new_name' (if provided
+  # and found), otherwise by 'name'.
+  #
+  # Once a script's 'new_name' has been seeded into the database, the script file
+  # can then be renamed back and forth between its old name and its new_name (or to
+  # any other name), and the corresponding script row in the db will be renamed.
```

Comment:
I'm going to try to lay this out explicitly to make sure I understand the rename process:

State | file name | `name` | `new_name` | Script is accessible via
-------|--------------|-----------|------------------|---------------------
0 | csp1.script | csp1 | nil | `get_without_cache('csp1')`
1 | csp1.script | csp1 | csp1-2017 | `get_without_cache('csp1')` or `get_without_cache('csp1-2017')`
2 | csp1-2017.script | csp1-2017 | csp1-2017 | `get_without_cache('csp1-2017')`

Getting from state 0 to 1 is straightforward, it's just setting a new property on `csp1`.

To get from state 1 to 2, the script will by found by it's `new_name` in the query on line 828 (note that its `name` won't match `options[:name]` until the script is re-seeded). Then the `update!()` on line 831 will update the script's name, and we'll be back to having just one canonical name for the script. Once you're in that state, you could remove the `new_name` from the script.

Think it's worth writing a test for that second step?

Reply:
Thanks for writing this out, Ram ! This table looks right. This PR (step 2 in the dev doc) gets us through State 1 from your table. As an additional clarification, it's important to start redirecting csp1 --> csp1-2017 (step 3 in the dev doc) and let it reach production, before we go to your state 2 (step 4 in the dev doc), so that we are never serving up blanks on /s/csp1.